scipy/integrate/_ivp/tests/test_ivp.py

```diff
@@ -980,3 +980,135 @@ def test_integration_zero_rhs(method):
     assert_(result.success)
     assert_equal(result.status, 0)
     assert_allclose(result.y, 1.0, rtol=1e-15)
+
+
+@pytest.mark.parametrize('method', ['Radau'])
+def test_mass_matrix_ODE(method):
+    """ The idea is to test the "mass" option with a simple vector ODE
+    on the array y=(y0,y1,y2,...,yn), which reads:
+      d(y[k])/dt = eigvals[k]*y, with eigvals[k] the k-th system eigenvalue
+      <=> dy/dt = diag(eigvals)*y
+    This simple ODE (no mass matrix) can also be reformulated as a:
+      M*dy/dt = y,  with M=diag(1/eigvals).
+    This allows to verify that the mass matrix implementation is coherent."""
+
+    rtol = atol = 1e-10
+    n = 7  # number of variables
+    eigvals = np.array([float(i)*(-1)**(i) for i in range(1, n+1)])
+    mass = diags(diagonals=(1/eigvals,), offsets=(0,), format='csc')
+
+    # solve both ODEs
+    tf = 1.0  # final time
+    y0 = np.ones((n,))
+    true_solution = y0*np.exp(eigvals*tf)
+    sol_with_mass = solve_ivp(fun=lambda t, x: x, t_span=(0., tf),
+                              y0=y0, max_step=np.inf, rtol=rtol, atol=atol,
+                              jac=None, jac_sparsity=None, method=method,
+                              vectorized=False, first_step=None, mass=mass)
+    sol_without_mass = solve_ivp(fun=lambda t, x: eigvals*x, t_span=(0., tf),
+                                 y0=y0, max_step=np.inf, rtol=rtol, atol=atol,
+                                 jac=None, jac_sparsity=None, method=method,
+                                 vectorized=False, first_step=None, mass=None)
+
+    assert_(sol_without_mass.success,
+            msg=f'solver {method} failed without mass matrix')
+    assert_(sol_with_mass.success,
+            msg=f'solver {method} failed with mass matrix')
+    assert_allclose(sol_without_mass.y[:, -1], true_solution,
```

Comment:
Good that you've used `assert_allclose`. That is preferred over `assert_almost_equal` and similar.